.github/workflows/flowzone.yml

```diff
@@ -18,3 +18,23 @@ jobs:
     with:
       docker_images: |
         ghcr.io/balena-io/deploy-to-balena-action
+
+  test-deploy-to-cloud:
+    name: Test deploy to balenaCloud
+    runs-on: ubuntu-latest
+    needs: flowzone
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Replace the build image and Dockerfile for testing
+        run: |
+          yq e -i '.runs.image = "docker://ghcr.io/balena-io/deploy-to-balena-action:${{ github.event.pull_request.head.sha }}"' \
+            action.yml
+          echo FROM busybox > Dockerfile
```

Comment:
Might be worth a comment here to explain why the Dockerfile is being overwritten.